SU2_CFD/src/solver_direct_turbulent.cpp

```diff
@@ -2952,6 +2966,279 @@ void CTurbSASolver::BC_NearField_Boundary(CGeometry *geometry, CSolver **solver_
   //
 }
 
+void CTurbSASolver::SetNuTilde_FW(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics,
+                                           CNumerics *visc_numerics, CConfig *config, unsigned short val_marker) {
+  
+  /*--- Local variables ---*/
+  
+  unsigned short iDim, jDim, iVar, iNode;
+  unsigned long iVertex, iPoint, iPoint_Neighbor, counter;
+  
+  su2double Wall_HeatFlux, func, func_prim;
+  su2double *Normal, Area;
+  su2double div_vel, UnitNormal[3];
+  su2double **grad_primvar, tau[3][3];
+  su2double Vel[3], VelNormal, VelTang[3], VelTangMod, VelInfMod, WallDist[3], WallDistMod;
+  su2double Lam_Visc_Normal, Kin_Visc_Normal, dypw_dyp, Eddy_Visc, nu_til_old, nu_til_4, nu_til, cv1_3;
+  su2double T_Normal, P_Normal, M_Normal, Density_Normal;
+  su2double Density_Wall, T_Wall, P_Wall, Lam_Visc_Wall, Tau_Wall, Tau_Wall_Old;
+  su2double *Coord, *Coord_Normal;
+  su2double diff, Delta;
+  su2double U_Tau, U_Plus = 0.0, Gam = 0.0, Beta = 0.0, Phi, Q = 0.0, Y_Plus_White = 0.0, Y_Plus;
+  su2double TauElem[3], TauNormal, TauTangent[3], WallShearStress;
+  su2double Gas_Constant = config->GetGas_ConstantND();
+  su2double Cp = (Gamma / Gamma_Minus_One) * Gas_Constant;
+  
+  unsigned short max_iter = 100;
+  su2double tol = 1e-10;
+
+  /*--- Get the freestream velocity magnitude for non-dim. purposes ---*/
+  
+  su2double *VelInf = config->GetVelocity_FreeStreamND();
+  VelInfMod = 0.0;
+  for (iDim = 0; iDim < nDim; iDim++)
+    VelInfMod += VelInf[iDim];
+  VelInfMod = sqrt(VelInfMod);
+  
+  /*--- Compute the recovery factor ---*/
+  // su2double-check: laminar or turbulent Pr for this?
```

Comment:
I think it should be the turbulent Pr.

Reply:
According to the paper is the laminar Pr.